Sources/NIO/Codec.swift

```diff
@@ -89,6 +88,32 @@ public protocol ByteToMessageDecoder: ChannelInboundHandler where InboundIn == B
     func shouldReclaimBytes(buffer: ByteBuffer) -> Bool
 }
 
+extension ByteToMessageDecoder {
+    public func decoderRemoved(ctx: ChannelHandlerContext) {
+    }
+
+    public func decoderAdded(ctx: ChannelHandlerContext) {
+    }
+
+    public func shouldReclaimBytes(buffer: ByteBuffer) -> Bool {
+        return true
+    }
+
+    public func decodeLast(ctx: ChannelHandlerContext, buffer: inout ByteBuffer) throws -> DecodingState {
```

Comment:
As a separate note, I will say that if we *do* want to have this we can get the different behaviours here by doing this:

```swift
extension ByteToMessageDecoder {
    public func decodeLast(ctx: ChannelHandlerContext, buffer: inout ByteBuffer) throws -> DecodingState {
        return .needMoreData
    }
}

extension ByteToMessageDecoder where InboundOut == ByteBuffer {
    public func decodeLast(ctx: ChannelHandlerContext, buffer: inout ByteBuffer) throws -> DecodingState {
        ctx.fireChannelRead(NIOAny(buffer))
        return .needMoreData
    }
}
```

Reply:
yeah we could but the whole design that we throw it under the bus unless it's ByteBuffer isn't good I think. We should make the user do smth sensible

Reply:
Deferring this for follow-up work.